pkg/sentry/fsimpl/fuse/connection.go

```diff
@@ -54,24 +76,258 @@ type FutureResponse struct {
 // Connection is the struct by which the sentry communicates with the FUSE server daemon.
 type Connection struct {
 	fd *DeviceFD
+
+	// MaxRead size in bytes
+	MaxRead uint32
+
+	// MaxWrite size in bytes
+	MaxWrite uint32
+
+	// MaxPages is the maximum number of pages for a single request to use
+	MaxPages uint16
+
+	// MaxBackground is the maximum number of outstanding background requests
+	MaxBackground uint16
+
+	// CongestionThreshold for number of background requests
+	CongestionThreshold uint16
+
+	// NumberBackground requests currently
+	NumBackground uint16
+
+	// ActiveBackground requests number currently queued for userspace
+	ActiveBackground uint16
+
+	// TODO: BgQuque
+	// some queue for background queued requests
+
+	// BgLock protects:
+	// MaxBackground, CongestionThreshold, NumBackground,
+	// ActiveBackground, BgQueue, Blocked
+	BgLock sync.Mutex
+
+	// Initialized if INIT reply has been received.
+	// Until it's set, FUSE request allocation will be suspended
+	Initialized bool
+
+	// protects Initialized
+	initializedLock sync.Mutex
+
+	// Blocked when:
+	// before the INIT reply is received,
+	// and if there are too many outstading backgrounds requests
+	Blocked bool
+
+	// Connected if connection established.
+	// Unset when umount, connection abort and device release
+	Connected bool
+
+	// Aborted via sysfs
+	Aborted bool
+
+	// ConnError if connection failed (version mismatch).
+	// Only set in INIT,
+	// before any other request,
+	// never unset.
+	// Cannot race with other flags.
+	ConnError bool
+
+	// ConnInit if connection successful.
+	// Only set in INIT.
+	ConnInit bool
+
+	// AsyncRead if read pages asynchronously.
+	// Only set in INIT.
+	AsyncRead bool
+
+	// AbortErr is true when FUSE will return an unique read error after abort.
+	// Only set in INIT.
+	AbortErr bool
+
+	// AtomicOTrunc is true when FUSE does not send a separate SETATTR request
+	// before open with O_TRUNC flag.
+	AtomicOTrunc bool
+
+	// ExportSupport is true if Filesystem supports NFS exporting.
+	// Only set in INIT.
+	ExportSupport bool
+
+	// WritebackCache is true if using write-back cache policy,
+	// false if using write-through policy>
+	WritebackCache bool
+
+	// ParallelDirops is true if allowing lookup and readdir in parallel,
+	// false if serialized.
+	ParallelDirops bool
+
+	// HandleKillpriv if fs handls killing suid/sgid/cap on write/chown/trunc.
+	HandleKillpriv bool
+
+	// CacheSymlinks if cache READLINK responses in page cache.
+	CacheSymlinks bool
+
+	/* Setting races on the following optimization-purpose flags are safe */
+
+	// NoOpen if open/release not implemented by fs
+	NoOpen bool
```

Comment:
There are a few settings like this that aren't being set during init or used yet. Should we remove them from here until they are needed?

Reply:
Let's leave them there and clean up eventually. It can both avoid the potential git conflict if multiple people need to add the same variables and avoid the duplication or inconsistency if not conflicted.